PR: https://github.com/odoo/odoo/pull/205542

From: 164b55c324e63c45339088823e3d55dfe2147b61
From: Einar Andersen (eian)
Date: 2025-08-22 10:53:17

Breaking data model changes scores: del:3 + add:0, change matches:
-    user_id = fields.Many2one('res.users', "Recruiter",
-    currency_id = fields.Many2one("res.currency", related="company_id.currency_id", readonly=True)
-    compensation = fields.Monetary(currency_field="currency_id")

Total Changes: 13

[IMP] hr_recruitment: skills v2

This PR introduces new and improved behaviours for managing skills on
models related to the recruitment application. These changes include:
* The ability to add expected skills to a job position.
* A new widget for managing the expected skills on a job position
* Improved dialog for creating skills both on an applicant and job position.
* Improved form view for job positions
* Improved form view for applicants
* Added a new field, score, on education degrees for better job position
  match calculation
* A new field to see how well an applicant matches a job position.
* The ability to browse matching positions from an applicant.
* A new cron to create activities for employees that are missing skills
  that are expected on a job position.
* A new report that allows you to explore how well employees match a job
  position in respect to their skills.
* Other small details like improved demo data, improved views, etc.

task-4680302

closes odoo/odoo#205542

Related: odoo/enterprise#83225
Related: odoo/upgrade#7858
Signed-off-by: Bertrand Dossogne (bedo) <bedo@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_recruitment/models/hr_job.py
+++ b/addons/hr_recruitment/models/hr_job.py
@@ -48,10 +48,6 @@ class HrJob(models.Model):
     manager_id = fields.Many2one(
         'hr.employee', related='department_id.manager_id', string="Department Manager",
         readonly=True, store=True)
-    user_id = fields.Many2one('res.users', "Recruiter",
-        domain="[('share', '=', False), ('company_ids', 'in', company_id)]", default=lambda self: self.env.user,
-        tracking=True, help="The Recruiter will be the default value for all Applicants in this job \
-            position. The Recruiter is automatically added to all meetings with the Applicant.")
     document_ids = fields.One2many('ir.attachment', compute='_compute_document_ids', string="Documents", readonly=True)
     documents_count = fields.Integer(compute='_compute_document_ids', string="Document Count")
     employee_count = fields.Integer(compute='_compute_employee_count')
@@ -62,8 +58,7 @@ class HrJob(models.Model):
     interviewer_ids = fields.Many2many('res.users', string='Interviewers', domain="[('share', '=', False), ('company_ids', 'in', company_id)]", tracking=True, help="The Interviewers set on the job position can see all Applicants in it. They have access to the information, the attachments, the meeting management and they can refuse him. You don't need to have Recruitment rights to be set as an interviewer.")
     extended_interviewer_ids = fields.Many2many('res.users', 'hr_job_extended_interviewer_res_users', compute='_compute_extended_interviewer_ids', store=True)
     industry_id = fields.Many2one('res.partner.industry', 'Industry', tracking=True)
-    currency_id = fields.Many2one("res.currency", related="company_id.currency_id", readonly=True)
-    compensation = fields.Monetary(currency_field="currency_id")
+    expected_degree = fields.Many2one("hr.recruitment.degree")
 
     activity_count = fields.Integer(compute='_compute_activities')
 

--- a/addons/hr_recruitment/models/hr_recruitment_degree.py
+++ b/addons/hr_recruitment/models/hr_recruitment_degree.py
@@ -1,4 +1,3 @@
-# -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from odoo import fields, models
@@ -9,9 +8,14 @@ class HrRecruitmentDegree(models.Model):
     _description = "Applicant Degree"
 
     name = fields.Char("Degree Name", required=True, translate=True)
+    score = fields.Float("Score", required=True, default=0)
     sequence = fields.Integer("Sequence", default=1)
 
     _name_uniq = models.Constraint(
         'unique (name)',
         'The name of the Degree of Recruitment must be unique!',
     )
+    _score_range = models.Constraint(
+        'check(score >= 0 and score <= 1)',
+        'Score should be between 0 and 100%',
+    )
